agent/api/container/container.go

```diff
@@ -895,6 +896,64 @@ func (c *Container) ShouldCreateWithASMSecret() bool {
 	return false
 }
 
+// RequiresCredentialSpec checks if container needs a credential spec resource
+func (c *Container) RequiresCredentialSpec() bool {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+
+	if c.DockerConfig.HostConfig == nil {
+		return false
+	}
+
+	hostConfig := &dockercontainer.HostConfig{}
+	err := json.Unmarshal([]byte(*c.DockerConfig.HostConfig), hostConfig)
```

Comment:
If here we just return false, it means we may create the container with wrong information. For example, there is **credentialspec** in the hostConfig but unmarshal fails for some reason, and the function returns false here. Eventually, the container still cannot be correctly created. I think we can just return error here.